PhysicsTools/KinFitter/src/TKinFitter.cc

```diff
@@ -1282,16 +1282,18 @@ void TKinFitter::printMatrix(const TMatrixD &matrix, const TString& name) {
 
   for (Int_t iSheet = 0; iSheet < nCols; iSheet += colsPerSheet) {
     log << "\n"
-	<< "     |";
-    for (Int_t iCol = iSheet; iCol < iSheet+colsPerSheet && iCol < nCols; iCol++)
-      log << std::setw(8) << iCol << "    |";
-    log << "\n"
-	<< topbar << " \n";
+	 << "     |";
+    for (Int_t iCol = iSheet; iCol < nCols; iCol++) {
+      if (iCol < iSheet+colsPerSheet) log << std::setw(8) << iCol << "    |";
+      log << "\n"
```

Comment:
The curly brace for the `for` loop should be before this line to match the previous logic.